src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -84,14 +84,132 @@ private static IEnumerable<TextEdit> FormatRazor(FormattingContext context, Razo
+        {
+            // We're looking for a code block like this:
+            //
+            // @{
+            //     var x = 1;
+            // }
+            if (node is CSharpCodeBlockSyntax expliciteCode &&
+                expliciteCode.Children.First() is CSharpStatementSyntax statement &&
+                statement.Body is CSharpStatementBodySyntax csharpStatementBody)
+            {
+                var openBraceNode = csharpStatementBody.OpenBrace;
+                var codeNode = csharpStatementBody.CSharpCode;
+                var closeBraceNode = csharpStatementBody.CloseBrace;
+
+                // IndentationContext doesn't track the need for any additional indentation in this case (Both Razor and HTML desirred indentation would be "0" above).
+                // Probably because it thinks you're doing something like "@{ DateTime.Now }"
+                // Regardless we need to help things out and inform FormatBlock to give us an extra level.
+                var additionalIndentation = true;
+
+                return FormatBlock(context, source, openBraceNode, codeNode, closeBraceNode, edits, additionalIndentation);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryFormatComplexCSharpBlock(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            // complex situations like
+            // @{
+            //  void Method(){
+            //      @(DateTime.Now)
+            //  }
+            // }
+            if (node is CSharpRazorBlockSyntax csharpRazorBlock &&
+                csharpRazorBlock.Parent is CSharpCodeBlockSyntax innerCodeBlock &&
+                csharpRazorBlock.Parent.Parent is CSharpCodeBlockSyntax outerCodeBlock)
+            {
+                var codeNode = csharpRazorBlock;
+                var openBraceNode = outerCodeBlock.Children.PreviousSiblingOrSelf(innerCodeBlock);
+                var closeBraceNode = outerCodeBlock.Children.NextSiblingOrSelf(innerCodeBlock);
+
+                return FormatBlock(context, source, openBraceNode, codeNode, closeBraceNode, edits, additionalIndentation: false);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryFormatHtmlInCSharp(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            // void Method()
+            // {
+            //     <div></div>
+            // }
+            if (node is MarkupBlockSyntax markupBlockNode &&
+                markupBlockNode.Parent is CSharpCodeBlockSyntax cSharpCodeBlock)
+            {
+                var previousChild = cSharpCodeBlock.Children.PreviousSiblingOrSelf(markupBlockNode);
```

Comment:
```suggestion
                var openBraceNode = cSharpCodeBlock.Children.PreviousSiblingOrSelf(markupBlockNode);
```

And can change `nextChild` to be `closeBraceNode`

Reply:
`openBraceNode` is not _technically_ accurate, for example it could be 
```razor
var x = t;
<div></div>
var t = x;
```
At this level Razor is agnostic as to the content of the C# (except within the IndentationContext).

Reply:
But you assign the previous/next to the open/close brace down below without modification